opengever/maintenance/scripts/contact_removal.py

```diff
@@ -0,0 +1,86 @@
+"""
+This script is used to remove contacts and related dossier participations.
+
+bin/instance run ./scripts/contact_removal.py
+
+optional arguments:
+  -n : dry-run.
+  -s : site root (used if multiple plone sites exists).
+"""
+
+from ftw.upgrade.progresslogger import ProgressLogger
+from opengever.dossier.behaviors.dossier import IDossierMarker
+from opengever.dossier.participations import PloneParticipationHandler
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_plone
+from plone import api
+import argparse
+import logging
+import sys
+import transaction
+
+logger = logging.getLogger('opengever.maintenance')
+handler = logging.StreamHandler(stream=sys.stdout)
+handler.setFormatter(logging.Formatter('%(levelname)s %(message)s'))
+logging.root.addHandler(handler)
+logging.root.setLevel(logging.INFO)
+
+
+class ContactRemover(object):
+
+    def run(self):
+        self.remove_contacts()
+        self.remove_contact_participations()
+
+    def remove_contacts(self):
+        catalog = api.portal.get_tool('portal_catalog')
+        brains = catalog.unrestrictedSearchResults(
+            portal_type='opengever.contact.contact')
+        contacts = [brain.getObject() for brain in brains]
+
+        logger.info("Found {} contacts to remove".format(len(contacts)))
+        api.content.delete(objects=contacts)
+
+    def remove_contact_participations(self):
+        catalog = api.portal.get_tool('portal_catalog')
+        brains = catalog.unrestrictedSearchResults(
+            object_provides=IDossierMarker.__identifier__)
+        for brain in ProgressLogger(
+                'Remove dossier contact participations.', brains):
+            handler = PloneParticipationHandler(brain.getObject())
+            participations = handler.get_participations()
+            for participation in participations:
```

Comment:
In case this loop is inefficient we could do this directly in the annotations to avoid popping elements one by one from the dictionary:
```
IAnnotations(self.context)['participations'] = PersistentDict({key:value for key, value in IAnnotations(self.context)['participations'] if not key.startswith(u'contact:')})
```

Reply:
My idea was that the official API also notfiy events, but will adjust if the script is to slow.